src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -781,27 +829,56 @@ private void loadTable(TableId tableId, TableBlob blob, boolean directLoadToTarg
     JobInfo jobInfo = JobInfo.newBuilder(loadJobConf)
       .setJobId(jobId)
       .build();
-    Job loadJob = BigQueryUtils.createBigQueryJob(bigQuery, jobInfo);
-    Job completedJob = loadJob.waitFor();
+    return BigQueryUtils.createBigQueryJob(bigQuery, jobInfo);
+  }
+
+  private String getLoadJobType(boolean directLoadToTarget) {
+    return directLoadToTarget ? LOAD_TARGET_JOB_TYPE : LOAD_STAGING_JOB_TYPE;
+  }
+
+  private void mergeStagingTable(TableId stagingTableId, TableBlob blob,
+                                 int attemptNumber) throws InterruptedException, IOException, DeltaFailureException {
+
+    LOG.info("Merging batch {} for {}.{} {}", blob.getBatchId(), blob.getDataset(), blob.getTable(),
+             attemptNumber > 0 ? "attempt: " + attemptNumber : "");
+
+    Job mergeJob = null;
+    if (attemptNumber > 0) {
```

Comment:
can we refactor this to a separate function and reuse here and above ( line 747) as well ?

Reply:
Done